[NNC] Pass on tensor names to intermediate tensors

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64961

Summary: Currently, NNC loops use hard-coded tensor names. This PR
passes the tensor names from the torchscript Value object. This can
facilitate and aid debugging for large graphs.

Test Plan: WIP

anijain2305 committed:
[NNC] Pass on tensor names to intermediate tensors

Summary: Currently, NNC loops use hard-coded tensor names. This PR
passes the tensor names from the torchscript Value object. This can
facilitate and aid debugging for large graphs.

Test Plan: WIP

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -2503,7 +2628,8 @@ Tensor TensorExprKernel::computeValue(const torch::jit::Value* v) {
     return custom_lowering(argInputs, outputShape, outputType, device_);
   }
-  return computeOperandValue(op, argInputs, outputShape, outputType, device_);
+  return computeOperandValue(
+      op, argInputs, outputShape, outputType, device_, name);
```
How about we set the name of the buffer returned by `computeOperandValue` here? That would mean that we only need to change one place.

anijain2305 replied:
Haha, that is literally just 1 line change. Thanks for pointing that out!

ZolotukhinM commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -2467,9 +2574,27 @@ c10::optional<ScalarType> findDtypeForValue(const torch::jit::Value* v) {
   return c10::nullopt;
 }
 
+static bool isValidIdentifierChar(char c, size_t pos) {
```
This will probably conflict with #65010. NBD if your PR lands first, but otherwise you'd have to rebase.

anijain2305 replied:
Thats completely ok. No hurry. If you have any suggestions for adding tests for my PR, plz let me know.

ZolotukhinM reviewed (approved):
Overall looks good, thanks for doing this! I only have one suggestion how to achieve a similar effect with a smaller change - please see inline :)

anijain2305 committed:
Update on "[NNC] Pass on tensor names to intermediate tensors"


Summary: Currently, NNC loops use hard-coded tensor names. This PR
passes the tensor names from the torchscript Value object. This can
facilitate and aid debugging for large graphs.

Test Plan: WIP

[ghstack-poisoned]

anijain2305 committed:
Update on "[NNC] Pass on tensor names to intermediate tensors"


Summary: Currently, NNC loops use hard-coded tensor names. This PR
passes the tensor names from the torchscript Value object. This can
facilitate and aid debugging for large graphs.

Test Plan: WIP

[ghstack-poisoned]

anijain2305 commented:
The tests are failing because some of the tests have hardcoded strings that expect to see `aten_xxx`/`sum` in the resulting loop IR. Will take a look further.
